modules/aprs/handlers/yb-tokens/yb-tokens-apr-handler.ts

```diff
@@ -0,0 +1,146 @@
+import { Chain, PrismaPoolAprItem, PrismaPoolAprItemGroup, PrismaPoolAprType } from '@prisma/client';
+import { YbAprHandlers } from '../../../pool/lib/apr-data-sources/yb-apr-handlers';
+import { TokenApr, YbAprConfig } from './types';
+import { PoolAPRData, AprHandler } from '../../types';
+
+/**
+ * Helper utility functions for pool operations
+ */
+function collectsYieldFee(pool: PoolAPRData): boolean {
+    return pool.type === 'COMPOSABLE_STABLE' || pool.type === 'META_STABLE' || pool.type === 'PHANTOM_STABLE';
+}
+
+function tokenCollectsYieldFee(token: any): boolean {
+    return token.token.underlyingTokenAddress !== null && token.token.underlyingTokenAddress !== undefined;
+}
+
+/**
+ * Calculator for yield-bearing tokens APR
+ * This calculates the APR for various yield-bearing tokens in pools
+ */
+export class YbTokensAprHandler implements AprHandler {
+    private ybTokensAprHandlers: YbAprHandlers;
+
+    constructor(
+        private aprConfig: YbAprConfig,
+        chain: Chain,
+    ) {
+        this.ybTokensAprHandlers = new YbAprHandlers(this.aprConfig, chain);
```

Comment:
Wonder if we need to dissect the `YbAprHandlers` a bit as well. In a first run, I think we could copy them as-is and refactor them in a second phase.